### DISCUSSION TITLE
Log file with lua filter information

### DISCUSSION THREAD

**Discussion by estedeahora** [jgm/pandoc#9659] [Q&A] [answered]
I am using pandoc with a set of custom lua filters. In the call to pandoc I use '--log=file.log', but I am interested in this log file incorporating information arising from the lua filters. In the lua filter I display this information in the console via io.write or error.

Is there any way to incorporate this directly into the 'file.log'?

Thanks!

**Comment by jgm** [comment]
I don't believe there is.  @tarleb would it be feasible to expose a function that calls `report` and generates a special log message like FilterWarning or FilterInfo?  I can't remember whether the kind of threading of state we do would allow this.

**Comment by tarleb** [comment, marked as answer]
Using [warn](https://www.lua.org/manual/5.4/manual.html#pdf-warn) should generate a pandoc log entry (since pandoc 3.1.4). Console output can then be controlled via `warn('@on')` and `warn('@off')`.

**Reply by estedeahora** [reply]
Thank you very much for your answer, but it doesn't seem to be working with 'warn'. Any idea what I might be doing wrong?

**Reply by tarleb** [reply]
What's your pandoc version? This was added in pandoc 3.1.4, so maybe an update might make this work?

**Reply by estedeahora** [reply]
I'm using Pandoc 3.1.12.1. 

I tried again and managed to solve it with 'warn'! I don't know what I did wrong yesterday. 

Thank you very much!

**Reply by estedeahora** [reply]
Would it be possible to extend this behaviour to 'error' and 'io.write'?

**Reply by tarleb** [reply]
Maybe we could provide a `report` function as suggested by @jgm. This would give more general access to pandoc's logging system. There are other issues, like #9077, which could be solved by allowing finer-grained access to logging objects. Feel free to raise a separate issue for this.
